src/buffer.ts

```diff
@@ -206,6 +207,26 @@ export class WriteMessageBuffer {
     return this;
   }
 
+  writeHeaders(headers: MessageHeaders): this {
+    if (this.messagePos < 0) {
+      throw new BufferError("cannot writeHeaders: no current message");
+    }
+    const entries = Object.entries(headers).filter(
+      ([_, value]) => value !== undefined
+    ) as Array<[keyof typeof HeaderCodes, string | Buffer]>;
+    this.buffer.writeUInt16(entries.length);
+    for (const [code, value] of entries) {
+      this.buffer.writeUInt16(HeaderCodes[code]);
+      if (Buffer.isBuffer(value)) {
+        this.buffer.writeUInt32(value.byteLength);
+        this.buffer.writeBuffer(value);
+      } else {
+        this.buffer.writeString(value);
```

Comment:
I'd add an assertion `typeof value === 'string'` and raise an error if it's not a string and not a buffer.